python/caches/hydrotools/caches/hdf.py

```diff
@@ -0,0 +1,108 @@
+"""
+=================================
+HDF5-backed Caching Functionality
+=================================
+Cache pandas objects resulting from long processes for later retrieval to avoid 
+reprocessing.
+
+Classes
+---------
+ - HDFCache
+
+"""
+
+import pandas as pd
+from typing import Callable
+
+class HDFCache:
```

Comment:
This looks great, just curious why the `*args` `**kwargs`? It seems more appropriate to include all the args [`HDFStore`](https://github.com/pandas-dev/pandas/blob/ea05559dc748eafe7de4f3ea39401b2d3f21bb4e/pandas/io/pytables.py#L556) includes.

Reply:
Additionally, it would be nice to include the `__del__` method for the garbage collector in the case of using `HDFCache` without a context manager.

Reply:
My approach to `*args` and `**kwargs` was to limit explicit parameters to those required (`path`) and those where I change the default (`format`). The reason is allow flexibility if more parameters get added to `pandas.HDFStore` later. Additionally, `HDFStore` itself passes arguments on to pytables, so I figured was easier to limit explicit parameters to those essential to the functioning of `HDFCache`.

Roger that on `__del__`.

Reply:
Yeah and that's a valid point. Maybe we can lean on the [`forge`](https://github.com/dfee/forge) library to automagically bring over those type hints. Its yet another dep to bring in, but im using it over in `_restclient` and it really small, so I am not overly concerned about bring it in to the project. Thoughts? To your point, I hope that future versions of python include decorators or some way to extend/share type hints more easily. `inspect` is not cutting it at the moment.

Cool cool 👍

Reply:
It seems like the issue with wrapping functions is a contentious one without a one-size-fits-all solution.

https://hynek.me/articles/decorators/

Reply:
Yeah, decorators are just so flexible that I would assume conversations quickly dissolve into users returning to their opinion corner instead of actually coming to a solution of sorts. Nice article though, it seems that `functools` has addressed many of the issues raised in that article -- which is good. However, using decorators to extend function args almost like you would in an inherited class or mixin still has not been agreed upon enough to make its way into the standard library.